src/sicks3000.cc

```diff
@@ -310,3 +310,355 @@ unsigned short SickS3000::CreateCRC(uint8_t *Data, ssize_t length)
   }
   return CRC_16;
 }
+
+/*----------------------------------------------------------------------------
+** Error Diagnostics
+*---------------------------------------------------------------------------*/
+#define SEG_0   0x00
+#define SEG_1_  0xB0
+#define SEG__1  0x86
+#define SEG_2   0xDB
+#define SEG_3   0xCF
+#define SEG_4   0xE6
+#define SEG_5   0xED
+#define SEG_6   0xFD
+#define SEG_7   0x87
+#define SEG_8   0xFF
+#define SEG_9   0xEF
+#define SEG_d   0xDE
+#define SEG_E   0xF9
+#define SEG_F   0xF1
+#define SEG_G   0xBD
+#define SEG_H   0xF6
+#define SEG_L   0xB8
+#define SEG_n   0x54
+#define SEG_o   0xDC
+#define SEG_P   0xF3
+#define SEG_u   0x9C
+#define SEG_y   0xEE
+
+#define SEVEN_SEGMENT_CODE(first, second) (first + (second << 8))
+
+#define NUM_S3000_ERROR_CODES 63
+
+typedef struct
+{
+  const uint32_t error;
+  const char* cause;
+  const char* suggested_action;
+  
+} ErrorCodeLUTEntry;
+
+const ErrorCodeLUTEntry s3000_errors[1] = {
+  {SEVEN_SEGMENT_CODE(SEG_1_,SEG_0), "Object in protective field (1_.)", "No error. The status indication eases system testing on the use of simultaneous protective fields or in an EFI system (if the OSSDs on the guest are not used, a protective field infringement is not signaled via the red LED as required in the standard)."}
+};
```

Comment:
Consider dropping the macro code and just having fields for `first_char`, `second_char`.

Drop the array length specifier— you don't need it.